src/twisted/internet/fdesc.py

```diff
@@ -10,9 +10,11 @@
 import os
 import errno
 try:
-    import fcntl
+    import fcntl as _fcntl
 except ImportError:
     fcntl = None
+else:
+    fcntl = _fcntl
```

Comment:
What's going on with this pattern? It's in a few places

Reply:
I took this pattern from here: https://github.com/python/mypy/issues/1297#issuecomment-508593494

I couldn't figure out a cleaner way to eliminate the mypy error, other than doing:

```
try:
    import fcntl
except ImportError:
    fcntl = None    # type: ignore
```

Maybe it would be better to do this?

```
from twisted.python.reflect import requireModule

fcntl = requireModule("fcntl")
```

Reply:
In the `requireModule` approach, would mypy still be able to use type hints/stubs for the imported module? My guess is it wouldn't, in which case this approach would reduce the reach of the type checking.

Reply:
I don't know how to specify type hints for a module.  I've looked for examples and asked on https://gitter.im/python/typing but haven't found an example.

Reply:
I don't know if it's even possible to attach a package name to a dynamically imported module in a way that mypy understands. I tried adding `assert fcntl.__name__ == 'fcntl'`, but that doesn't change the way mypy sees the module.

The `fcntl = _fcntl` pattern you used might be the best option, for now.

Reply:
Well if I do this from the repl:

```
>>> import fcntl
>>> type(fcntl)
<class 'module'>
```

I couldn't figure out a way to add an annotation for `class 'module'`

Reply:
The annotation for "a module" is `types.ModuleType`. However, that tells mypy that it's dealing with an arbitrary module, not one specific module for which it might know the type signatures of its contents.

Reply:
To clarify things with an example:
```py
from types import ModuleType
from twisted.python.reflect import requireModule

import fcntl as direct
indirect: ModuleType = requireModule('fcntl')

reveal_type(direct)
reveal_type(indirect)

direct.flock()
indirect.flock()
```
For this code, mypy outputs:
```
testcase.py:7:13: note: Revealed type is '_importlib_modulespec.ModuleType'
testcase.py:8:13: note: Revealed type is '_importlib_modulespec.ModuleType'
testcase.py:10:1: error: Too few arguments for "flock"  [call-arg]
testcase.py:11:1: error: Module has no attribute "flock"  [attr-defined]
```
So while the revealed type is equal, a direct import results in mypy knowing that `flock` exists and what its signature is, while it does not have that knowledge if the the import is done via `requireModule`.

Reply:
How would you add type annotations for the return type of `requireModule` here: https://github.com/twisted/twisted/blob/4e42bf4928266b5135be6d66f1532d52317d8731/src/twisted/python/reflect.py#L179  ?

Reply:
The best I can think of is:
```py
from types import MethodType
from typing import TypeVar, Union
T = TypeVar('T')
def requireModule(name: str, default: T = None) -> Union[MethodType, T]:
```
That will not make it easy to annotate any code using `requireModule`, but a better solution might not exist.